src/java.base/share/classes/java/lang/annotation/ElementType.java

```diff
@@ -26,106 +26,213 @@
+ * as part of a type parameter or local variable declaration:
  *
- * <p>The constants {@link #ANNOTATION_TYPE}, {@link #CONSTRUCTOR}, {@link
- * #FIELD}, {@link #LOCAL_VARIABLE}, {@link #METHOD}, {@link #PACKAGE}, {@link
- * #MODULE}, {@link #PARAMETER}, {@link #TYPE}, and {@link #TYPE_PARAMETER}
- * correspond to the declaration contexts in JLS {@jls 9.6.4.1}.
+ * {@snippet id='example' :
+ * @Target({ElementType.TYPE_PARAMETER, ElementType.LOCAL_VARIABLE})
+ * public @interface MyAnnotation {}
+ * }
  *
- * <p>For example, an annotation whose interface is meta-annotated with
- * {@code @Target(ElementType.FIELD)} may only be written as a modifier for a
- * field declaration.
+ * <h2>Declaration annotations and type-use annotations</h2>
  *
- * <p>The constant {@link #TYPE_USE} corresponds to the type contexts in JLS
- * {@jls 4.11}, as well as to two declaration contexts: class and interface
- * declarations (including annotation declarations) and type parameter
- * declarations.
+ * <p>Most annotations in Java code are <b>declaration
+ * annotations</b>, which act like modifiers of declarations (such as
+ * a field declaration). The constants of this class cover all ten
+ * kinds of annotatable declarations, plus a subcategory of {@link
+ * #TYPE} called {@link #ANNOTATION_TYPE}. An annotation interface can
+ * be used as a declaration annotation if it either omits {@code
+ * @Target}, or uses it to list which specific kinds of declarations
+ * it should apply to.
  *
- * <p>For example, an annotation whose interface is meta-annotated with
- * {@code @Target(ElementType.TYPE_USE)} may be written on the class or
- * interface of a field (or within the class or interface of the field, if it
- * is a nested or parameterized class or interface, or array class), and may
- * also appear as a modifier for, say, a class declaration.
+ * <p>There are also <b>type-use annotations</b> (sometimes called
+ * "type annotations"), which can appear anywhere a Java type is being
+ * indicated (normally, immediately preceding that type). To enable
+ * use as a type-use annotation, an annotation interface must
+ * explicitly include {@link #TYPE_USE} in its list of targets. 
  *
- * <p>The {@code TYPE_USE} constant includes class and interface declarations
- * and type parameter declarations as a convenience for designers of
- * type checkers which give semantics to annotation interfaces. For example,
- * if the annotation interface {@code NonNull} is meta-annotated with
- * {@code @Target(ElementType.TYPE_USE)}, then {@code @NonNull}
- * {@code class C {...}} could be treated by a type checker as indicating that
- * all variables of class {@code C} are non-null, while still allowing
- * variables of other classes to be non-null or not non-null based on whether
- * {@code @NonNull} appears at the variable's declaration.
+ * <h3 id="ambiguous">Ambiguous contexts</h3>
+ *
+ * <p>For six kinds of declarations, type-use annotations can also
```

Comment:
Same number assertion remark; pattern bindings most likely will have this same problem too.